tests/client_test.py

```diff
@@ -610,3 +630,317 @@ def test_intercept_unary_unary_failed_request(self):
+        with mock.patch('logging.config.dictConfig'):
+            mock_exception = self._get_mock_transport_exception()
+            interceptor = self._create_test_interceptor()
+            result = interceptor._get_fault_message(mock_exception)
+            self.assertEqual(result, self._MOCK_TRANSPORT_ERROR_MESSAGE)
+
+    def test_get_fault_message_unknown_failure(self):
+        with mock.patch('logging.config.dictConfig'):
+            mock_exception = {}
+            interceptor = self._create_test_interceptor()
+            result = interceptor._get_fault_message(mock_exception)
+            self.assertEqual(result, None)
+
+
+class ExceptionInterceptorTest(TestCase):
+    """Tests for the google.ads.googleads.client.ExceptionInterceptor class."""
+
+    _MOCK_FAILURE_VALUE = b"\n \n\x02\x08\x10\x12\x1aInvalid customer ID '123'."
+
+    def _create_test_interceptor(self):
+        return google.ads.google_ads.client.ExceptionInterceptor()
+
+    def test_init_(self):
+        interceptor = self._create_test_interceptor()
+        self.assertEqual(interceptor._RETRY_STATUS_CODES,
+            (grpc.StatusCode.INTERNAL, grpc.StatusCode.RESOURCE_EXHAUSTED))
+
+    def test_get_request_id(self):
+        mock_metadata = (('request-id', '123456'),)
+        interceptor = self._create_test_interceptor()
+        result = interceptor._get_request_id(mock_metadata)
+        self.assertEqual(result, '123456')
+
+    def test_get_request_id_no_id(self):
+        mock_metadata = (('another-key', 'another-val'),)
+        interceptor = self._create_test_interceptor()
+        result = interceptor._get_request_id(mock_metadata)
+        self.assertEqual(result, None)
+
+    def test_get_google_ads_failure(self):
+        interceptor = self._create_test_interceptor()
+        mock_metadata = ((interceptor._FAILURE_KEY, self._MOCK_FAILURE_VALUE),)
+        result = interceptor._get_google_ads_failure(mock_metadata)
+        self.assertIsInstance(result, error_protos.GoogleAdsFailure)
+
+    def test_get_google_ads_failure_decode_error(self):
+        interceptor = self._create_test_interceptor()
+        mock_failure_value = self._MOCK_FAILURE_VALUE + b'1234'
+        mock_metadata = ((interceptor._FAILURE_KEY, mock_failure_value),)
+        result = interceptor._get_google_ads_failure(mock_metadata)
+        self.assertEqual(result, None)
+
+    def test_get_google_ads_failure_no_failure_key(self):
+        mock_metadata = (('another-key', 'another-val'),)
+        interceptor = self._create_test_interceptor()
+        result = interceptor._get_google_ads_failure(mock_metadata)
+        self.assertEqual(result, None)
+
+    def test_get_google_ads_failure_with_None(self):
+        interceptor = self._create_test_interceptor()
+        result = interceptor._get_google_ads_failure(None)
+        self.assertEqual(result, None)
+
+    def test_handle_grpc_exception(self):
+        """If exception is not retryable and is a GoogleAdsFailure convert to
```

Comment:
**Nitpick**: This isn't formatted correctly for a multi-line docstring.

https://www.python.org/dev/peps/pep-0257/#id17

Reply:
Fixed.